environment/src/main/java/com/sequenceiq/environment/network/v1/converter/EnvironmentBaseNetworkConverter.java

```diff
@@ -97,6 +98,14 @@ protected Set<DeploymentRestriction> getDeploymentRestrictionForPrivateSubnet(Su
                 : DeploymentRestriction.NON_ENDPOINT_ACCESS_GATEWAYS;
     }
 
+    protected boolean isAllSubnetsPublic(CreatedCloudNetwork createdCloudNetwork) {
+        return createdCloudNetwork.getSubnets().stream().allMatch(e -> e.isPublicSubnet());
+    }
+
+    protected  Set<DeploymentRestriction> getDeploymentRestrictionWhenPublicSubnet(CreatedCloudNetwork createdCloudNetwork) {
+        return isAllSubnetsPublic(createdCloudNetwork) ? DeploymentRestriction.ALL : DeploymentRestriction.ENDPOINT_ACCESS_GATEWAYS;
```

Comment:
Can we have dedicated logs here what value is calculated by the elvis?
I would rather use an if condition with logs for easier investigation from the logs in the future.